src/utils/update/beneficial_owners_managing_officers_data_fetch.ts

```diff
@@ -34,29 +36,65 @@ const initialiseBoAndMoUpdateAppData = (appData: ApplicationData) => {
 };
 
 export const retrieveBeneficialOwners = async (req: Request, appData: ApplicationData) => {
+  const transactionId = appData.transaction_id;
+  const overseasEntityId = appData.overseas_entity_id;
   const pscs: CompanyPersonsWithSignificantControl = await getCompanyPsc(req, appData[EntityNumberKey] as string);
-  if (pscs) {
-    for (const psc of (pscs.items || [])) {
-      logger.info("Loaded psc " + psc.kind);
-      if (psc.ceasedOn === undefined) {
-        if (psc.kind === "individual-beneficial-owner") {
-          const individualBeneficialOwner = mapPscToBeneficialOwnerTypeIndividual(psc);
-          logger.info("Loaded individual Beneficial Owner " + individualBeneficialOwner.id + " is " + individualBeneficialOwner.first_name + ", " + individualBeneficialOwner.last_name);
-          appData.update?.review_beneficial_owners_individual?.push(individualBeneficialOwner);
-        } else if (psc.kind === "corporate-entity-beneficial-owner") {
-          const beneficialOwnerOther = mapPscToBeneficialOwnerOther(psc);
-          logger.info("Loaded Beneficial Owner Other " + beneficialOwnerOther.id + " is " + beneficialOwnerOther.name);
-          appData.update?.review_beneficial_owners_corporate?.push(beneficialOwnerOther);
-        } else if (psc.kind === "legal-person-beneficial-owner") {
-          const beneficialOwnerGov = mapPscToBeneficialOwnerGov(psc);
-          logger.info("Loaded Beneficial Owner Gov " + beneficialOwnerGov.id + " is " + beneficialOwnerGov.name);
-          appData.update?.review_beneficial_owners_government_or_public_authority?.push(beneficialOwnerGov);
-        }
+
+  if (!pscs || pscs.items?.length === 0) {
```

Comment:
We could add a test for this scenario, example: 'should not set BO data in appData if no company pscs returned'

Reply:
or should not get BO private data if

Reply:
Done.